src/Grids/grid_utils.jl

```diff
@@ -175,7 +175,7 @@ constant grid spacing `Δ`, and interior extent `L`.
 
 # Grid domains
 @inline domain(topo, N, ξ) = CUDA.@allowscalar ξ[1], ξ[N+1]
-@inline domain(::Flat, N, ξ::AbstractArray) = CUDA.@allowscalar ξ[1], ξ[1]
+@inline domain(::Flat, N, ξ::AbstractArray) = ξ[1]
```

Comment:
wouldn't that fail without the `@allowscalar`?

Reply:
A one-element range is used to store the coordinate when we have a "located" `Flat` direction. Array / CuArray is only used for stretched coordinates. I can think of a use case but I'm inclined to say that we should cross that bridge when we get there...

Reply:
ok!